sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/policy/AuthenticationPolicy.java

```diff
@@ -36,32 +25,16 @@
      * @param scopes the scopes used in credential, using default scopes when empty
      */
     public AuthenticationPolicy(TokenCredential credential, AzureEnvironment environment, String... scopes) {
-        Objects.requireNonNull(credential);
-        this.credential = credential;
+        super(credential, scopes);
         this.environment = environment;
-        this.scopes = scopes;
     }
 
     @Override
-    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        if ("http".equals(context.getHttpRequest().getUrl().getProtocol().toLowerCase(Locale.ROOT))) {
-            return Mono.error(new RuntimeException("token credentials require a URL using the HTTPS protocol scheme"));
-        }
-
-        Mono<AccessToken> tokenResult;
-        if (this.scopes == null || this.scopes.length == 0) {
-            String defaultScope = ResourceManagerUtils.getDefaultScopeFromRequest(
-                context.getHttpRequest(), environment);
-            tokenResult = this.credential.getToken(new TokenRequestContext().addScopes(defaultScope));
-        } else {
-            tokenResult = this.credential.getToken(new TokenRequestContext().addScopes(scopes));
+    protected String[] getScopes(HttpPipelineCallContext context, String[] scopes) {
+        if (CoreUtils.isNullOrEmpty(scopes)) {
+            scopes = new String[1];
+            scopes[0] = ResourceManagerUtils.getDefaultScopeFromRequest(context.getHttpRequest(), environment);
```

Comment:
Dynamic `scopes` in azure-resourcemanager-resources.

It depends on `AccessTokenCache` to retrieve new token when scopes changed.

Reply:
Yeah the AccessTokenCache will retreive a new token if a different scope is passed in.